src/openforms/js/components/admin/form_design/logic/actions/dmn/DMNActionConfig.js

```diff
@@ -24,32 +25,40 @@ const ERRORS = {
 };
 
 const DecisionDefinitionIdField = () => {
-  const {values, setFieldValue, getFieldProps, errors, touched} = useFormikContext();
+  const {
+    values: {pluginId},
+    setFieldValue,
+    getFieldProps,
+    errors,
+    touched,
+    setFieldTouched,
+    handleChange,
+  } = useFormikContext();
   const [field] = useField('decisionDefinitionId');
-  const [decisionDefinitions, setDecisionDefinitions] = useState([]);
 
-  useAsync(async () => {
-    if (!values.pluginId) {
-      setFieldValue('decisionDefinitionId', '');
-      setDecisionDefinitions([]);
-      return;
-    }
+  const {error, value: decisionDefinitions = []} = useAsync(async () => {
+    if (!pluginId) return [];
 
     const response = await get(DMN_DECISION_DEFINITIONS_LIST, {
-      engine: values.pluginId,
+      engine: pluginId,
     });
 
-    const definitionChoices = response.data.map(item => [item.id, item.label]);
-    setDecisionDefinitions(definitionChoices);
+    return response.data.map(item => [item.id, item.label]);
+  }, [pluginId, setFieldValue]);
```

Comment:
```suggestion
  }, [pluginId]);
```
this is not used in the hook, is it?